### PR TITLE
795 refactoring of hs_lables app models

### PR DESCRIPTION
**Pull request by pkdash**
Need to be reviewed and merged for AGU release.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by alvacouch** [comment]
I have checked this and it fully checks out as complete according to my code review of the new My Resources page yesterday. I am willing to pass it but perhaps someone other than I should look at it.

File: hs_labels/models.py [hunk 1]
```diff
@@ -6,173 +6,109 @@
...
 
 
-####################################
-# privileges are numeric codes 1-4
-####################################
-class LabelCodes(object):
+class UserResourceLabels(models.Model):
+    """ Labels of a user for a resource
+    This model stores labels of an individual user, like an access
+    control list; T
```
Comment (hyi) [inline hs_labels/models.py]: remove the last character T in the docstring

**Comment by pkdash** [comment]
@hyi If you have finished reviewing, I can make push this change.

**Comment by hyi** [comment]
@pkdash still reviewing, will let you know after I am done.

**Comment by pkdash** [comment]
@hyi sounds good.

File: hs_labels/tests/test_resource_labelling.py [hunk 2]
```diff
@@ -110,15 +112,6 @@ def test01labels(self):
...
 
-    def test02folders(self):
-        cat = self.cat  # user
-        scratching = self.scratching  # resource
-        cat.ulabels.file_resource(scratching, "penalty clause")
-        self.assertTrue(match_lists(cat.ulabels.filed_resources, [scratching]))
-        self.assertTrue(match_lists(cat.ulabels.resources_of_interest, [scratching]))
-        self.assertTrue(match_lists(scratching.rlabels.get_folder(cat), 'penalty clause'))
-        self.assertTrue(match_lists(scratching.rlabels.get_users(), [cat]))
-
```
Comment (hyi) [inline hs_labels/tests/test_resource_labelling.py]: @pkdash after these test functions are removed, not sure whether we need to change the following test functions to be sequential, e.g., change test03favorite() to test02favorite(), etc., just to be clearer, but these are very trivial things and don't have to be done at all. It is up to you.

**Comment by hyi** [comment]
@pkdash I am done with reviewing and don't see anything wrong. I am giving +1 here. Just let me know whether you want to leave it as is to be merged or address a couple of very minor comments and push it quickly. Just let me know and I will merge it and deploy to beta after I hear from you.

**Comment by pkdash** [comment]
@hyi I will make these changes and let you know. Thanks for the quick review.

**Comment by pkdash** [comment]
@hyi Once the Jenkins check is all clear, you can merge this PR.

**Pull request merged**